app/src/test/java/com/xmartlabs/template/helper/GsonRequiredTest.java

```diff
@@ -0,0 +1,117 @@
+package com.xmartlabs.template.helper;
+
+import com.google.gson.ExclusionStrategy;
+import com.google.gson.FieldAttributes;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.xmartlabs.template.service.gson.GsonExclude;
+import com.xmartlabs.template.service.gson.GsonRequired;
+import com.xmartlabs.template.service.gson.JsonRequiredFieldException;
+import com.xmartlabs.template.service.gson.RequiredFieldDeserializer;
+
+import junit.framework.Assert;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import lombok.Builder;
+import lombok.Data;
+
+public class GsonRequiredTest {
```

Comment:
CircleCI is running, and since this repo is public i think is a better option than Jenkins:
https://circleci.com/gh/xmartlabs/Android-Base-Project/125?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link